L1t stage1 unpacker 75x

mulhearn opened the pull request:
This PR brings the L1T packer/unpacker tool in line with (not yet available) AMC13 firmware.  It is necessary for data taking with the Stage 1 trigger.  The changes to HLT configs are for demonstration purposes only, showing the needed config changes for one example.  HLT experts will need to make these changes properly in the configuration framework.  Because of a backwards incompatibility between the actual firmware and the offline anticipation of the firmware, additional customization will be needed to unpack L1 payload from RAW data produced via MC under 74X, if such capability is needed.

dburns7 committed:
importing L1 DPG Ntuples code

dburns7 committed:
replaced file paths containing DPG, compiles

dburns7 committed:
merged upgrade formats

dburns7 committed:
added crab3 example config for NanoDST

jbsauvan committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

jbsauvan committed:
Stage-2: Fix tower ratio decompression

Xtian9 committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

Xtian9 committed:
MP/Demux unpacker/emulator updates as of Easter '15 pattern tests

Xtian9 committed:
MP/Demux unpacker/emulator updates as of Easter '15 pattern tests

Xtian9 committed:
MP/Demux updates to match Andy's firmware changes

Xtian9 committed:
MP/Demux updates to match Andy's firmware changes

Xtian9 committed:
fix y-axis orientation in MP energy sums

Xtian9 committed:
fix y-axis orientation in MP energy sums

jimbrooke committed:
add example test config

jimbrooke committed:
add raw file

Xtian9 committed:
make MP header frames configurable

Xtian9 committed:
make MP header frames configurable

dburns7 committed:
added Extra and Ntuple examples

Xtian9 committed:
update mp/demux unpacker/emulator

Xtian9 committed:
update mp/demux unpacker/emulator

Xtian9 committed:
update sum order in demux

Xtian9 committed:
update sum order in demux

Xtian9 committed:
update cordic to calculate negative ET sum

Xtian9 committed:
update cordic to calculate negative ET sum

matz-e committed:
Add access to bits [15:8] of MP7 link headers.

Implemented via getMask() and complementary arguments in the constructor.

matz-e committed:
Dump link/block content when unpacking in debug mode.

jimbrooke committed:
put back data directory and test files

jimbrooke committed:
add reader for AMC central buffer dump

jimbrooke committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into mydevel-branch

matz-e committed:
Rename mask → capID for MP7 link headers.

jimbrooke committed:
add central buffer test file

matz-e committed:
Print block header with proper with when debugging.

jimbrooke committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

jimbrooke committed:
add test files

jimbrooke committed:
add AMC13 dump converter

matz-e committed:
Account for proper AMC headers and trailers.

Raw → Digi for now.  Data checks pending, too.

matz-e committed:
Store the AMC header and trailer in the AMC and block classes.

matz-e committed:
Improve output formatting in the unpacking framework.

matz-e committed:
Add unpacking testing configuration using cosmics data.

jimbrooke committed:
add utility for unpacking data

jimbrooke committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into mydevel-branch

matz-e committed:
Improve unpacking error message when no FED data found.

matz-e committed:
Improve unpacking debug output.

matz-e committed:
Fix payload size calculation when unpacking.

matz-e committed:
Skip filler words in the MP7 Payload, not the framework.

matz-e committed:
Reset cout back to dec after printing hex.

matz-e committed:
Fix up unpackers for CapID usage.

Merges the RCTEMUnpacker and RCTRegionUnpacker, preserving their unpacking
logic.  Unpacking methods are switched based on the CapID of the block.

matz-e committed:
Add missing unpacking framework files.

jimbrooke committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

jimbrooke committed:
move utils to utils

jimbrooke committed:
update with more options

jimbrooke committed:
fix text dump

jimbrooke committed:
print to log not file

jimbrooke committed:
include input data

jimbrooke committed:
add filter for fat events

jimbrooke committed:
add fat event filter

matz-e committed:
Add some trailer checks.

jbsauvan committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into jbsauvan-l1t-devel-CMSSW_7_5_0_pre1

jbsauvan committed:
Add getter/setter for e/g calibration/identification type and version

jbsauvan committed:
Stage-2 clustering: Fix sharing

jbsauvan committed:
Stage-2: Add compression in e/g calibration and identification

matz-e committed:
Only check trailer CRC for AMC13 blocks when valid.

jimbrooke committed:
add skip events functionality

jimbrooke committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into mydevel-branch

matz-e committed:
Whitespace.

matz-e committed:
Account for header/trailer subtraction in the AMC classes.

Commit:
add new HI L1 menu for testing

matz-e committed:
Write AMC header and trailer to AMC13 payloads.

matz-e committed:
Write CRC for AMC trailers properly.

jimbrooke committed:
add prtection if ECAL or HCAL are missing, for stage 1 unpacker testing

jimbrooke committed:
add RCT analyzer for region and EG input histograms

jimbrooke committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into mydevel-branch

jimbrooke committed:
quiet zero rank RCT EM

jimbrooke committed:
quiet zero rank objects in print out

jimbrooke committed:
add debug option

ginnocen committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

jimbrooke committed:
add thresholds

ginnocen committed:
removing ordering for the RCT/EM objects

ginnocen committed:
Merge branch 'l1t-devel-CMSSW_7_5_0_pre1' of github.com:cms-l1t-offline/cmssw into mydevel-branch-FixUnpacker

mulhearn committed:
Merged l1t-devel-CMSSW_7_5_0_pre1 from repository cms-l1t-offline

Conflicts:
	EventFilter/L1TRawToDigi/src/implementations_stage1/CaloSetup.cc
	EventFilter/L1TRawToDigi/src/implementations_stage1/RCTEMUnpacker.cc
	EventFilter/L1TRawToDigi/src/implementations_stage1/RCTRegionUnpacker.cc
	EventFilter/L1TRawToDigi/test/test_DigiToRawToDigi.py
	EventFilter/L1TRawToDigi/utils/unpackBuffers-CaloStage1.py
	EventFilter/L1TRawToDigi/utils/unpackBuffers-CaloStage2.py
	L1Trigger/L1TCommon/python/customsPostLS1.py

mulhearn committed:
Merged l1t-proposed-CMSSW_7_5_0_pre5 from repository cms-l1t-offline

mulhearn committed:
Add unpacker/packer updates of Matthias,Jim,and Gian Michele

mulhearn committed:
Merged l1t-devel-CMSSW_7_5_0_pre5 from repository cms-l1t-offline

mulhearn committed:
add missing configs to customization for DigiToRaw

mulhearn committed:
Merged l1t-devel-CMSSW_7_5_0_pre5 from repository cms-l1t-offline

mulhearn committed:
fix the standard MC production sequence configs

mulhearn committed:
replace l1tRawToDigi for now, as it is needed by DQM

mulhearn committed:
Merged l1t-devel-CMSSW_7_5_0_pre5 from repository cms-l1t-offline

mulhearn committed:
demo changes needed for HLT configs

cmsbuild commented:
A new Pull Request was created by @mulhearn  for CMSSW_7_5_X.

L1t stage1 unpacker 75x

It involves the following packages:

EventFilter/L1TRawToDigi
HLTrigger/Configuration
L1Trigger/L1TCommon


The following packages do not have a category, yet:

EventFilter/L1TRawToDigi

@Martin-Grunewald, @perrotta, @cmsbuild, @mulhearn, @fwyzard can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @geoff-smith, @jalimena this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Martin-Grunewald commented:
Hi,

With a proper fillDescriptions() method, there is no need anymore for "existsAs" which
should thus be removed.

cmsbuild commented:
+1
Tested at: b9663c85e1c688fb82e4a9416bf70453b83ed60a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9908/10606/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/80a60d7129933cc9ab951d4339c28acfba5d8937
https://github.com/cms-sw/cmssw/commit/51433ada85ecc29bd41921564858ed847b7d5538
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9908/10606/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9908/10606/git-merge-result

matz-e committed:
Fix firmware version determination.

matz-e committed:
Remove misleading warnings.

The code was changed a while back to accomodate cable splitting, unpacking
with fixed positions.  The warnings pertain to old code, where the unpacked
quantities were just pushed back onto the collection.
